mitodl/mit-open
# add userlist bookmark button and add to user list modal

### What are the relevant tickets?
Closes https://github.com/mitodl/mit-open/issues/721

### Description (What does it do?)
This PR adds the bookmark button to the `LearningResourceCard` page component. Clicking this button displays the "Add to User List" modal. The `AddToListDialog` page component was modified to support adding a resource to `UserList` objects.

### Screenshots (if appropriate):
<img width="1916" alt="image" src="https://github.com/mitodl/mit-open/assets/12089658/0c8d7c51-4263-48ad-b3df-6e39a89bfd28">
<img width="1917" alt="image" src="https://github.com/mitodl/mit-open/assets/12089658/2f5dd254-6008-41a9-99e2-171acf040c00">
<img width="221" alt="image" src="https://github.com/mitodl/mit-open/assets/12089658/7bde06c4-0180-485b-b5dd-baddecb0b225">
<img width="220" alt="image" src="https://github.com/mitodl/mit-open/assets/12089658/4d61ffff-14b8-4fbd-8849-a1a622a2f026">

### How can this be tested?
 - Spin up `mit-open` locally on this branch
 - Visit the home page at http://localhost:8063
 - Click the bookmark icon on any of the courses under "Upcoming Courses"
 - This should bring up the "Add to User List" dialog
 - Click the "Create a new list" button at the bottom
 - This should open the "Create User List" dialog
 - Create a new User List by filling out the dialog fields and clicking Save
 - Check the box to save the Learning Resource to your User List
 - Click the X in the upper right to close the dialog
 - Click the bookmark icon on the same Learning Resource to bring the dialog back up
 - Ensure that the User List we added the Learning Resource to is still checked
 - Add a few more Learning Resources to your User List
 - Visit http://localhost:8063/userlists
 - Click the title of the User List you created and ensure that all the items you added are present
